aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -4039,4 +4039,39 @@ Tensor linalg_solve_triangular(
   return out;
 }
 
+Tensor linalg_vander(
+    const Tensor& x,
+    c10::optional<int64_t> N,
+    bool increasing) {
+  const auto x_ = x.dim() == 0 ? x.unsqueeze(-1) : x;
+
+  auto shape = x_.sizes().vec();
+  const auto n = N.value_or(shape.back());
+  TORCH_CHECK(n >= 0, "N must be non-negative.");
+
+  // Get the dtype of cumsum
+  auto dtype = x_.scalar_type();
+  auto options = x_.options()
+                   .dtype(at::isIntegralType(dtype, /*includeBool=*/true) ? kLong : dtype);
+
+  // n = 0 or n = 1 case (Empty or one row case)
+  shape.push_back(std::min<int64_t>(n, 1LL));
+  auto ones =  x_.new_ones(shape, options);
+
+  if (n <= 1) {
+    // new_ones does not propagate requires_grad. UGH
```

Comment:
Even thought the tensor options should have `requires_grad=True`?

Reply:
`tensor.options()` does not propagate `requires_grad` (??!!).

Reply:
What are we trying to achieve with this `requires_gradness`? Gradient won't be propagated back to x anyway, because there's no connection between new_ones and x.

Reply:
One (not particularly great) way of making sure that x gets 0 derivative in this case is to construct output as `x * 0 + 1` with some slicing to make sure sizes are correct.

Reply:
FWIW, `ones_like` does propagate grads.

Reply:
I agree with @ngimel, the all-ones column is always ones no matter what the input is, so there is no need need in making it require grad.

Reply:
We might want the gradient chain not break and make `x` receive all zero gradients, but I don't think what we have here accomplishes that.

Reply:
I see your point. Then we could select a single element from the tensor with, say, `index`, multiply it with 0, add 1 and then expand to the shape of `x`. Should be better than multiplying the whole `x` with zero.

Reply:
I needed this because otherwise gradchecks would complain that "no output is differentiable". I'm happy to hear waht's a good fix for this. thoughts @albanD?

Reply:
What about this?

```python
In [1]: import torch

In [2]: x = torch.rand(3, 3, requires_grad=True)

In [3]: y = x.pow(0.0)

In [4]: y.sum().backward()

In [5]: x.grad
Out[5]: 
tensor([[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]])

```

Reply:
my favorite is `y=x*0+1`, but both @nikitaved's and my option will require some slicing to make sure output is correct size. If we didn't support batch dimension it would be easier :-) Also, both options suffer from unexpected nan propagation, so maybe do `y=(x*0).fill_(1)` which is a bit of perf penalty but oh well.

Reply:
I think this idea could be potentially improved perf-wise if done on just a single element of `x` with proper `expand`.

Reply:
Well, on paper a "detached" graph is a valid way to represent a Tensor full of zeros.
It's just that we don't usually gradcheck functions that are independent of their input.

If we really want to do this, you can re-use some test function we have for gradcheck: `output = torch._C._functions.UndefinedGrad()(input)`. This function will just ignore the input and return undefined gradients (so gradients of zeros).
Note that not keeping the graph around is even better for perf though!

Reply:
Ok, let's just not check those inputs.

Reply:
Actually just setting `requires_grad=False` for those inputs does not work. I'm tentatively leaving it as it is, as:
- It passes the tests
- If you are using this function to generate a tensor full of ones, you have larger issues to worry about than the fact that we are tracking the derivatives for that leaf tensor you'll never use.

Reply:
But this is wrong, this is creating a dangling tensor with requires_grad that breaks the gradient propagation. Can we skip the tests instead? vander with n=1 is essentially non-differentiable and should be treated as such.

Reply:
Getting 0 gradient is different from not getting gradient (the chain is not broken in the former, and broken in the latter). In vander with n=1 we have the latter (x.grad will be None, if x is not leaf all nodes below x won't get gradients). I agree that vander should not be used to get tensor full of ones, but having workarounds in the code just to pacify the tests (and not have the "right" behavior, for however right is defined) is not the way. Returning ones that don't require grad is fine with me, and it should be fine with the tests.  Or it should be commented in the code "we know this is wrong and silly, but that's the easiest thing to do and there's no harm".

Reply:
After syncing with @albanD, we decided the simplest solution is to just disallow `N <= 1` for now. If a user asks for it, we'll add it then taking into account what they think it's a good design.

Reply:
This happened again with composite compliance for linalg.matrix_power, I would quite like it if we had a consolidated strategy for this